classes/gateways/class.pmprogateway_stripe.php

```diff
@@ -370,10 +373,48 @@ static function pmpro_payment_option_fields( $values, $gateway ) {
 					<?php
 				}
 				?>
-				<input type='hidden' name='stripe_connect_user_id' id='stripe_connect_user_id' value='<?php echo esc_attr( $values['stripe_connect_user_id'] ) ?>'/>
+				<input type='hidden' name='live_stripe_connect_user_id' id='live_stripe_connect_user_id' value='<?php echo esc_attr( $values['live_stripe_connect_user_id'] ) ?>'/>
 				<input type='hidden' name='live_stripe_connect_secretkey' id='live_stripe_connect_secretkey' value='<?php echo esc_attr( $values['live_stripe_connect_secretkey'] ) ?>'/>
-				<input type='hidden' name='test_stripe_connect_secretkey' id='test_stripe_connect_secretkey' value='<?php echo esc_attr( $values['test_stripe_connect_secretkey'] ) ?>'/>
 				<input type='hidden' name='live_stripe_connect_publishablekey' id='live_stripe_connect_publishablekey' value='<?php echo esc_attr( $values['live_stripe_connect_publishablekey'] ) ?>'/>
+			</td>
+        </tr>
+		<tr class="gateway gateway_stripe" <?php if ( $gateway != "stripe" ) { ?>style="display: none;"<?php } ?>>
+            <th scope="row" valign="top">
+                <label> <?php esc_attr_e( 'Stripe Connection (Sandbox):', 'paid-memberships-pro' ); ?></label>
+            </th>
+			<td>
+				<?php
+				$connect_url_base = apply_filters( 'pmpro_stripe_connect_url', 'https://connect.paidmembershipspro.com' );
+				if ( self::has_connect_credentials( 'sandbox' ) ) {
+					$connect_url = add_query_arg(
+						array(
+							'action' => 'disconnect',
+							'gateway_environment' => 'test',
```

Comment:
Can we pass the gateway into this too? We may need this same setup for other gateways (PayPal Commerce) in the future.

```suggestion
							'gateway' => 'stripe',
							'gateway_environment' => 'test',
```

Reply:
Yeah makes sense. Not sure how much of the server code will be reusable, but we can at least use the same server/url like this.